ui/src/pages/ShoeMap.tsx

```diff
@@ -19,21 +20,74 @@ const MapContainer = styled.div`
   align-items: center;
 `;
 
+function getPinState(id: string, selectedPin: string) {
+  switch (selectedPin) {
+    case "":
+      return PinState.Resting;
+    case id:
+      return PinState.Selected;
+    default:
+      return PinState.Unfocused;
+  }
+}
+
+const pins = [
+  { lat: 43.4723, lng: -80.5449 },
+  { lat: 43.48, lng: -80.5449 },
+];
+
 export const ShoeMap: React.FC = () => {
   const zoom = 12;
-  const currentLocation = { lat: 43.4723, lng: -80.5449 };
+  const history = useHistory();
+  const location = useLocation();
+  const [selectedPin, setSelectedPin] = React.useState("");
+
+  React.useEffect(() => {
+    const searchParams = new URLSearchParams(location.search);
+    const loc = searchParams.get("location");
+    if (loc) {
+      setSelectedPin(loc);
+    }
+  }, [location]);
+
+  const handlePopupOpen = React.useCallback(
+    (id: string) => {
+      return () => {
+        setSelectedPin(id);
+        history.push({
+          search: `?location=${id}`,
+        });
+      };
+    },
+    [setSelectedPin]
+  );
```

Comment:
maybe this is a noob question but do you know why we need the nested functions here?

Reply:
It's because its a curried function, check the how the prop is used by the `Pin` component.

Basically the Pin component calls the function with an id, and so we have to return a function that accepts an id to be used by the click handler.

Reply:
Ooh ya I didn't catch that :) Thanks!